src/controller/task.ts

```diff
@@ -0,0 +1,57 @@
+import { Request, Response, NextFunction } from 'express'
+import TaskModel, { Task } from '../model/task'
+
+export default {
+  async createTask (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const { title, description, importance } = req.body
+      const newTask: Task = await TaskModel.create({
+        title,
+        description,
+        importance
+      })
+      res.status(201).json(newTask)
+    } catch (err) {
+      next(err)
+    }
+  },
+  async getAllTasks (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const tasks: Task[] = await TaskModel.find()
+      res.status(200).json({
+        results: tasks.length,
+        tasks
+      })
+    } catch (err) {
+      next(err)
+    }
+  },
+  async getTaskById (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const task: Task | null = await TaskModel.findById(req.params.id)
```

Comment:
There should be 404 error if task is not found. Same for other methods where you search by id.

Reply:
I was thinking about moving null and not valid id checking to middleware (because this will be using in Get by id, Delete and Patch methods) , something like
```
export checkIdValidity= (req, res, next) =>{
    const id = req.params.id
    if(!mongoose.Types.ObjectId.isValid(id)) {
        return res.status(400).json({message: `Invalid ${id}`})
    }
    next();
```

Reply:
Yes, it's fine for validating mongo id. (later we will use different approach for this)

Reply:
But the not found part should be handled in the code. (also it's possible to create shared function with this logic, but the error middleware needs to handle those errors properly with correct status codes/messages)

Reply:
30afaebf219a08e95436d413662504fff035dd68